lib/sdn_epc/converter.ex

```diff
@@ -0,0 +1,107 @@
+defmodule SdnEpc.Converter do
+  require SdnEpc.OfpmRecord
+
+  @ofp_version Application.get_env(:sdn_epc, :ofp_version)
+
+  def convert({:packet_in, xid, body}) do
```

Comment:
From what I see it's always the second element in the tuple so there shouldn't be a problem with extracting it out like:

```elixir
def convert({type, xid,  body} = msg) do
   SdnEpc.OfpmRecord.ofp_message(
       version: @ofp_version,
       xid: xid, %% or elem(msg, 1),
       body: body(type, body))
end
```